enclave/certificate_gen.go

```diff
@@ -0,0 +1,85 @@
+package enclave
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	log "github.com/sirupsen/logrus"
+	"github.com/blk-io/crux/config"
+	"math/big"
+	"os"
+	"time"
+	"fmt"
+)
+
+func CertGen(configFile string) error {
+	config.CertInitFlags()
+	err := config.CertLoadConfig(configFile)
+	if err != nil {
+		log.Errorf("Loading from config failed with %v", err)
+		return err
+	}
+
+	serialNo, err := rand.Int(rand.Reader, big.NewInt(32))
+	if err != nil {
+		log.Errorf("Generating random number failed with %v", err)
+		return err
+	}
+	certificateParam := &x509.Certificate{
+		SerialNumber: serialNo,
+		Subject: pkix.Name{
+			Organization:  config.CertGetStringSlice(config.OrgName),
+			Country:       config.CertGetStringSlice(config.CountryCode),
+			Province:      config.CertGetStringSlice(config.Province),
+			Locality:      config.CertGetStringSlice(config.Locality),
+			StreetAddress: config.CertGetStringSlice(config.Address),
+			PostalCode:    config.CertGetStringSlice(config.PostalCode),
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(
+			config.CertGetInt(config.ValidityYears),
+			config.CertGetInt(config.ValidityMonths),
+			config.CertGetInt(config.ValidityDays)),
+		IsCA:                  config.CertGetBool(config.IsCA),
```

Comment:
If it's not a certificate authority, could we do signing on behalf of a local authority who's keys are on the same host? I appreciate this isn't exactly a production use case, but could be good to explore